features/AssetCategory.feature

```diff
@@ -0,0 +1,189 @@
+      | engineId       | "engine-ayse" |
+      | body.name      | "size"        |
+      | body.valueType | "integer"     |
+      | body.mandatory | false         |
+    When I successfully execute the action "device-manager/assetCategory":"linkMedatadata" with args:
+      | engineId       | "engine-ayse" |
+      | _id | "truck" |
+      | _metadataId | "size"  |
+    Then The document "engine-ayse":"asset-category":"truck" content match:
+      | name                        | "truck"   |
+      | assetMetadatas[0].name      | "size"    |
+      | assetMetadatas[0].valueType | "integer" |
+      | assetMetadatas[0].mandatory | false     |
+
+  Scenario: Link and unlink an asset and an AssetCategory
+    When I successfully execute the action "device-manager/asset":"linkCategory" with args:
+      | _id        | "tools-MART-linked" |
+      | categoryId | "truck"             |
+      | engineId   | "engine-ayse"       |
+    Then The document "engine-ayse":"assets":"tools-MART-linked" content match:
+      | categories | ['truck'] |
+    When I successfully execute the action "device-manager/asset":"unlinkCategory" with args:
+      | _id        | "tools-MART-linked" |
+      | categoryId | "truck"             |
+      | engineId   | "engine-ayse"       |
+    Then The document "engine-ayse":"assets":"tools-MART-linked" content match:
+      | categories | [] |
+
+  Scenario: Remove an AssetCategory and verify propagation to a linked asset
+    When I successfully execute the action "device-manager/assetCategory":"create" with args:
+      | engineId  | "engine-ayse" |
+      | body.name | "tool"        |
+    When I successfully execute the action "device-manager/asset":"linkCategory" with args:
+      | _id        | "tools-MART-linked" |
+      | categoryId | "tool"              |
+      | engineId   | "engine-ayse"       |
+    And I refresh the collection "engine-ayse":"assets"
+    And I successfully execute the action "device-manager/assetCategory":"delete" with args:
+      | engineId | "engine-ayse" |
+      | _id      | "tool"        |
+    Then The document "engine-ayse":"assets":"tools-MART-linked" content match:
+      | categories | [] |
+
+  Scenario: Update a metadata, and verify edition propagation
+    When I successfully execute the action "device-manager/metadata":"update" with args:
+      | engineId       | "engine-ayse" |
+      | _id            | "size"        |
+      | body.mandatory | true          |
+    Then The document "engine-ayse":"asset-category":"truck" content match:
+      | assetMetadatas[0].mandatory | true |
+
+
+  Scenario: Delete a metadata, and verify edition propagation
+    When I successfully execute the action "device-manager/metadata":"delete" with args:
+      | engineId | "engine-ayse" |
+      | _id      | "size"        |
+    Then The document "engine-ayse":"asset-category":"truck" content match:
+      | assetMetadatas | [] |
```

Comment:
Functional tests should be idempotent so we can play any of the scenario without having to rely on a state left by a previous scenario.

Usually we are cleaning the database between each step (see `features/support/hooks.js`) and recreating the state we need to play a particular scenario.

Reply:
Done